de1plus/plugins/web_api/plugin.tcl

```diff
@@ -77,5 +78,25 @@ proc ::wibble::togglePowerOff {state} {
     dict set response status 200
     dict set state response header content-type "" {application/json charset utf-8}
     dict set response content "{status: \"ok\"}"
+    sendresponse $response
+}
+
+proc ::wibble::checkStatus {state} {
+    if { ![check_auth $state] } {
+        return;
+    }
+
+    dict set response status 200
+    dict set state response header content-type "" {text/plain charset utf-8}
+    
+    # Returning a simple text 1 if the machine is in anything other than an idle state. Return text 0 if idle.
+    # Return values chosen by cribbing from Supereg/homebridge-http-switch
+
+    if { $::de1_num_state($::de1(state)) != "Sleep" } {
```

Comment:
Looking at the states in `machine.tcl`, you might want to decide which states are "status good", or just return the state name in text and let the caller decide which are "good" for what they want to do next. I don't know that you can, for example, interrupt `Descale` and some others.

If you do decide to digest several states into a Boolean, I've found the Tcl [`in`/`ni`](https://www.tcl.tk/man/tcl8.6/TclCmd/expr.htm#M15) operator useful.

Reply:
Hi jeff - nothing here interrupts, it simply returns whether the machine is sleeping or not. It's primarily intended for use so that a smarthome hub can know whether the device is on or off in the event it was controlled independently from the hub.